### PR TITLE
feat: v3

### PR DESCRIPTION
**Pull request by oterral**
BREAKING CHANGES: Make mbt closer to openlayers and maplibre api

# How to

Open the review app and test everything

# Others

- [x] It's not a hack or at least an unauthorized hack :).
- [x] The images added are optimized.
- [x] Everything in ticket description has been fixed.
- [x] The author of the MR has made its own review before assigning the reviewer.
- [x] The title means something for a human being and follows the [conventional commits](https://www.conventionalcommits.org/) specification.
- [x] The title contains [WIP] if it's necessary.
- [x] Labels applied. if it's a release? a hotfix?
- [x] The new class' members & methods are well documented
- [x] Tests added.

### CODE REVIEW

**Comment by oterral** [comment]
@friedjoff  @danji90   please take a look and tets the review app

**Review by friedjoff — APPROVED** [review]
Looks much better now overall! 🚀

Let me know if I should look at anything in more detail. 👀

File: doc/public/static/examples/ol-maplibre-layer.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    // Display the feature informations
+    document.getElementById('content').innerHTML = feature
+      ? JSON.stringify(feature.get('vectorTileFeature'), null, 2)
+      : 'No feature found';
+  });
+
+  // Toggle between 2 different styles
+  document.getElementById('change').onclick = () => {
+    layer.style = layer.style === 'travic_v2' ? 'aerial' : 'travic_v2';
```
[the remark this answers is not in the document]
Comment (oterral) [reply doc/public/static/examples/ol-maplibre-layer.js:39]: @danji90 fixed

File: src/ol/controls/RoutingControl.ts [hunk 2]
```diff
@@ -108,152 +111,120 @@ const getFlatCoordinatesFromSegments = (
...
     this.api = new RoutingAPI({
       ...options,
     });
 
-    /** @private */
     this.routingLayer =
       options.routingLayer ||
-      new RoutingLayer({
-        name: 'routing-layer',
+      new VectorLayer({
```
[the remark this answers is not in the document]
Comment (oterral) [reply src/ol/controls/RoutingControl.ts:223]: The ghost feature has been fixed for the bug it's a problem in ol so we can't use the proper hit detction for points, see https://github.com/openlayers/openlayers/pull/16393

**Pull request merged**